smac/smbo/smbo.py

```diff
@@ -126,12 +127,16 @@ def run(self):
 
             self.logger.debug("Intensify")
 
-            self.incumbent, inc_perf = self.intensifier.intensify(
-                challengers=challengers,
-                incumbent=self.incumbent,
-                run_history=self.runhistory,
-                aggregate_func=self.aggregate_func,
-                time_bound=max(0.01, time_spend))
+            try: self.incumbent, inc_perf = self.intensifier.intensify(
+                  challengers=challengers,
+                  incumbent=self.incumbent,
+                  run_history=self.runhistory,
+                  aggregate_func=self.aggregate_func,
+                  time_bound=max(0.01, time_spend))
+            except (TAEAbortException, BudgetExhaustedException):
```

Comment:
Is there any disadvantage of storing the incumbent in `self.intensifier.incumbent`? If handling the exceptions in intensification, it is more complicated to abort the smbo-loop.